src/app/static/src/app/components/plots/bar/utils.ts

```diff
@@ -17,9 +16,11 @@ export interface ErrorBars {
     }
 }
 
-export type ChartDataSetsWithErrors =  ChartDataset & {
-    errorBars?: ErrorBars
-}
+type ChartDataSetsWithErrors<Data = BarChartDefaultData> = BarChartDataset<Data> & { errorBars?: ErrorBars }
+
+type ChartDataWithErrors<Data = BarChartDefaultData> = Omit<ChartData<"bar", Data, string>, "datasets"> & { datasets: ChartDataSetsWithErrors<Data>[] }
```

Comment:
this is cool, this performs surgery, basically the `ChartData` type doesnt allow datasets to have this `errorBars` prop that we carry around in the `chartData` for our ease of use so I `Omit` the `datasets` prop from the normal `ChartData` type and add in our custom datasets type with `errorBars`, `ChartDataSetsWithErrors` type (while also exposing the `Data` generic so that we can specify the type of data)

Reply:
Typescript is crazy